pkg/metrics/logging_hook.go

```diff
@@ -15,6 +15,13 @@ import (
 
 var metricsInitialized chan struct{} = make(chan struct{})
 
+func FlushLoggingMetrics() {
```

Comment:
This looks like the same logic as https://github.com/cilium/cilium/blob/a5824bd93c4761880ddd389e1e0af4ef2886429d/pkg/metrics/cell.go#L16-L19 and might be prone to race conditions or ordering?

Reply:
Given the current hives structure, that snippet is called only from the agent, while `FlushLoggingMetrics` is called only from the operator, hence we are safe.
But yeah, it's kind of confusing. Not sure how to improve this, however (apart from adding a comment to `FlushLoggingMetrics` to clarify its intended usage).

Reply:
@chancez Yeah this depends on only being called once from either operator/agent once.

@pippolo84 Agreed, this isn't ideal, been meaning to start a discussion about a potential better solution in the future.

Reply:
To make this safe, I can probably make this safe to *accidentally* call multiple times concurrently.